theming.js

```diff
@@ -466,28 +467,47 @@ const Transparency = new Lang.Class({
     },
 
     _updateStyles: function() {
-        this._transparent_style = 'background-color: rgba(' +
-                                  this._backgroundColor + ',' + this._alpha + ');' +
-                                  'border-color: rgba(' +
-                                  this._backgroundColor + ',' + this._borderAlpha + ');' +
-                                  'transition-duration: 500ms;';
-        this._opaque_style = 'background-color: rgba(' +
-                             this._backgroundColor + ',' + '1);' +
-                             'border-color: rgba(' +
-                             this._backgroundColor + ',' + this._originalBorderAlpha + ');' +
-                             'transition-duration: 300ms;';
+        this._getAlphas();
+
+        this._transparent_style =
+            'background-color: rgba(' +
+            this._backgroundColor + ', ' + this._transparentAlpha + ');' +
+            'border-color: rgba(' +
+            this._backgroundColor + ', ' + this._transparentAlphaBorder + ');' +
+            'transition-duration: ' + this._transparentTransition + 'ms;';
+
+        this._opaque_style =
+            'background-color: rgba(' +
+            this._backgroundColor + ', ' + this._opaqueAlpha + ');' +
+            'border-color: rgba(' +
+            this._backgroundColor + ',' + this._opaqueAlphaBorder + ');' +
+            'transition-duration: ' + this._opaqueTransition + 'ms;';
     },
 
     setColor: function(color) {
         this._backgroundColor = color.red + ',' + color.green + ',' + color.blue;
         this._updateStyles();
     },
 
-    setAlpha: function(alpha, borderAlpha, originalBorderAlpha) {
-        this._alpha = alpha.toString();
-        borderAlpha = Math.round(borderAlpha*100)/100;
-        this._borderAlpha = borderAlpha.toString();
-        this._originalBorderAlpha = originalBorderAlpha.toString();
-        this._updateStyles();
+    _getAlphas: function() {
```

Comment:
Do we need this only for the styles defined in our stylesheet? Or is is necessary also to support third party themes?

Reply:
We need this for both:

- to retrieve our own `opaque` and `transparent` styles
- if a (3rd party) theme author includes his own `opaque` and `transparent` styles

Does it answer your question?

Reply:
The css classes might come with specific colors, and we might be overriding this already.

This was the whole problem since the beginning: is there a property e.g. `alpha: 0.8` that we can use? If not, we need the full `rgba()`.

Reply:
If the color comes from the theme it's fine, that's what the auhor wants. Doesn't it makes sense? I can try to preapare a patch and see if it work, as I might be missing something :)

Reply:
What I meant is, if the user sets a specific color in the settings, this color will be overridden by the theme.

Normally, even if the theme proposes a default color, the GUI lets the user set whatever color he wants.

Reply:
In any case, as you suggested elsewhere, we could just drop the whole `opaque`/`transparent` styles, and remove this machinery. I'm fine with that!